subprojects/core/src/main/java/org/gradle/api/internal/file/copy/PathNotationConverter.java

```diff
@@ -55,17 +54,10 @@ public void convert(Object notation, NotationConvertResult<? super String> resul
                 || notation instanceof Number
                 || notation instanceof Boolean) {
             result.converted(notation.toString());
-        } else if (notation instanceof Closure) {
-            final Closure closure = (Closure) notation;
-            final Object called = closure.call();
-            convert(called, result);
         } else if (notation instanceof Callable) {
             final Callable callableNotation = (Callable) notation;
             final Object called = uncheckedCall(callableNotation);
```

Comment:
❓ shouldn't this use `DeferredUtil`?

Reply:
That would change the behaviour for `Provider<T>` values.

Reply:
Right. It would add support for `Provider<T>` where `T` is supported. I think we want it on `PathNotationConverter` for uniformity and to support e.g.:
```kotlin
copy {
    from { "foo/dir" }
    into { "bar/dir" }
}
```

Can be considered out of this PR's scope and done as a subsequent step for https://github.com/gradle/kotlin-dsl/issues/1077.